document.go

```diff
@@ -26,16 +26,42 @@ type Document struct {
 }
 
 // Add a VerificationMethod as AssertionMethod
+// If the controller is not set, it will be set to the documents ID
 func (d *Document) AddAssertionMethod(v *VerificationMethod) {
```

Comment:
VerificationMethod field is a []*VerificationMethods. This allows us to compare pointers to prevent duplicats.
It is now possible to do this:
```
vm := NewVerificationMetdhod()
doc.AddAssertionMethod(vm)
doc.AddAuthenticationMethod(vm)
assert.Equal(len(doc.VerificationMethod) == 1)
```

Reply:
Why not just use ID then? Because if you accept a pointer you can also get strange stuff like this:

```
vm := VerificationMethod{}
vm.ID = "foobar"

doc.AddAuthentication(&vm)
vm.Type = "some-type"
doc.AddAuthentication(&vm)
```
Is it added once?

or;

```
vm := &VerificationMethod{}
vm.ID = "first-vm"

doc.AddAuthentication(vm)
vm.ID = "second-vm"
doc.AddAuthentication(vm)
```

Second isn't added because the pointers are the same.


Having pointers in function signatures make assumptions on how the caller handles (memory manangement; reuse of struct instance for "optimization" or just because) the stuff that's passed into the function AROUND it, which is bad practice imo, which leads to strange bugs.